docs/DeveloperGuide.md

```diff
@@ -5,9 +5,231 @@
+
+This functionality is facilitated by `ParserSleep`. It implements one operation, namely:
+- `ParserSleep#parseSleepInput(String input)`
+
+This operation is exposed in the `SleepList` class as `SleepList#addSleep(String)`.
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `sleep add 7.5 d/2022-01-02` in the terminal,
+  the string is sent to `SleepList#addEntry(String)`, which calls `ParserSleep#parseSleepInput(String)`.
+
+- Step 2: Using `String.split()`, the method extracts information such as the duration and date of entry.
+
+- Step 3: It will create a new entry of class `SleepEntry` that extends `Entry`based on the information.
+
+- Step 4: The created `SleepEntry` instance is added into the `ArrayList<Entry> sleepList` attribute of the `SleepList`.
+
+
+### Sleep list feature
+
+The `sleep list` feature lists out the record of all the sleep data that the user has keyed in. The sleep data are all stored into a `ArrayList<Entry> sleepList` attribute of the `SleepList` Class. Sleep data are printed when the `printSleepList()` function is called.
+
+The `printSleepList()` function iterates through the `sleepList` and each entry will call `SleepEntry#toString()` to return its information string to be printed.
+
+The Sequence diagram for Sleep list feature is shown below. Unrelated attributes and Classes were excluded.
+![SleepListSeqDiagram](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepListSeqDiagram.jpg)
+
+
+### Sleep delete feature
+
+The `sleep delete` feature can delete the sleep record at specific index of sleep list. This functionality is facilitated by `SleepList`. It implements one operation, namely:
+- `deleteSleep(String line)`
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `sleep delete INDEX` in the terminal, the string is sent to `Ui#handleUserInput()`, which will call `Ui#handleSleepInput()`.
+
+- Step 2: After the `Ui#handleSleepInput()` matching `sleep delete` key word, the string will be passed into deleteSleep(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The sleep record (`Entry`) stored in the `ArrayList<Entry> sleepList` will be deleted by calling `sleepArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `SleepListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest sleep list will be updated to saving file by calling `SleepList#updateFile()`.
+
+The Sequence diagram for Sleep delete feature is shown below:
+![SleepDeleteDiagram.jpg](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepDeleteSeqDiagram.jpg)
```

Comment:
The image of the sequence diagram is not rendered correctly. 
![Screenshot 2024-04-04 152229](https://github.com/nus-cs2113-AY2324S2/tp/assets/110890669/3dc6420c-3f72-4b7b-8af9-7c7425d6bc95)